File: .agent/skills/mobile-design/scripts/mobile_audit.js
#!/usr/bin/env node
/**
 * Mobile Design Audit - Stub Script
 * Referenced by: /mobile workflow
 * 
 * Checks mobile app projects for:
 * - Touch target sizes (44x44 minimum)
 * - Color contrast ratios
 * - Platform guidelines compliance
 */

const fs = require('fs');
const path = require('path');

function auditProject(projectPath) {
    const results = {
        projectPath,
        issues: [],
        warnings: [],
        passed: []
    };

    if (!fs.existsSync(projectPath)) {
        console.error(`Error: Path '${projectPath}' does not exist`);
        process.exit(1);
    }

    // Detect platform
    const platforms = [];
    if (fs.existsSync(path.join(projectPath, 'android')) || fs.existsSync(path.join(projectPath, 'app/build.gradle'))) {
        platforms.push('android');
    }
    if (fs.existsSync(path.join(projectPath, 'ios')) || fs.readdirSync(projectPath).some(f => f.endsWith('.xcodeproj'))) {
        platforms.push('ios');
    }
    if (fs.existsSync(path.join(projectPath, 'package.json'))) {
        try {
            const pkg = JSON.parse(fs.readFileSync(path.join(projectPath, 'package.json'), 'utf8'));
            const deps = { ...pkg.dependencies, ...pkg.devDependencies };
            if (deps['react-native']) platforms.push('react-native');
            if (JSON.stringify(pkg).includes('flutter')) platforms.push('flutter');
        } catch (e) { }
    }
    if (fs.existsSync(path.join(projectPath, 'pubspec.yaml'))) {
        platforms.push('flutter');
    }

    if (platforms.length > 0) {
        results.passed.push(`✅ Detected platforms: ${[...new Set(platforms)].join(', ')}`);
    } else {
        results.warnings.push('⚠️ Could not detect mobile platform');
    }

    // Check for accessibility files
    const a11yIndicators = [
        "accessibility", "a11y", "contentDescription",
        "accessibilityLabel", "semanticLabel"
    ];

    function getFiles(dir, exts) {
        let files = [];
        const items = fs.readdirSync(dir, { withFileTypes: true });
        for (const item of items) {
            if (item.name.startsWith('.') || item.name === 'node_modules') continue;
            const res = path.resolve(dir, item.name);
            if (item.isDirectory()) {
                files = [...files, ...getFiles(res, exts)];
            } else if (exts.some(ext => item.name.endsWith(ext))) {
                files.push(res);
            }
        }
        return files;
    }

    const sourceFiles = getFiles(projectPath, ['.tsx', '.jsx', '.dart', '.kt', '.swift']);

    let hasA11y = false;
    for (const file of sourceFiles.slice(0, 20)) {
        try {
            const content = fs.readFileSync(file, 'utf8');
            if (a11yIndicators.some(ind => content.includes(ind))) {
                hasA11y = true;
                break;
            }
        } catch (e) { }
    }

    if (hasA11y) {
        results.passed.push("✅ Accessibility attributes detected");
    } else {
        results.warnings.push("⚠️ No accessibility attributes found in sampled files");
    }

    // Check for touch targets
    const touchIndicators = ["minHeight: 44", "minWidth: 44", "hitSlop", "touchableOpacity"];
    let hasTouchTargets = false;
    for (const file of sourceFiles.slice(0, 20)) {
        try {
            const content = fs.readFileSync(file, 'utf8');
            if (touchIndicators.some(ind => content.includes(ind))) {
                hasTouchTargets = true;
                break;
            }
        } catch (e) { }
    }

    if (hasTouchTargets) {
        results.passed.push("✅ Touch target sizing considerations found");
    } else {
        results.warnings.push("⚠️ Consider adding minimum touch target sizes (44x44)");
    }

    if (results.issues.length === 0) {
        results.passed.push("✅ Mobile design audit complete");
    }

    return results;
}

function printResults(results) {
    console.log("\n" + "=".repeat(50));
    console.log("📱 MOBILE DESIGN AUDIT");
    console.log("=".repeat(50));
    console.log(`Project: ${results.projectPath}\n`);

    if (results.issues.length > 0) {
        console.log("❌ ISSUES:");
        results.issues.forEach(i => console.log(`  ${i}`));
        console.log();
    }
    if (results.warnings.length > 0) {
        console.log("⚠️ WARNINGS:");
        results.warnings.forEach(w => console.log(`  ${w}`));
        console.log();
    }
    if (results.passed.length > 0) {
        console.log("✅ PASSED:");
        results.passed.forEach(p => console.log(`  ${p}`));
        console.log();
    }

    if (results.issues.length > 0) {
        console.log("❌ AUDIT FAILED");
        process.exit(1);
    } else {
        console.log("✅ AUDIT PASSED");
        process.exit(0);
    }
}

const args = process.argv.slice(2);
if (args.length < 1) {
    console.log("Usage: node mobile_audit.js <project_path>");
    process.exit(1);
}

printResults(auditProject(args[0]));

File: .agent/skills/mobile-design/scripts/mobile_audit.py
#!/usr/bin/env python3
"""
Mobile Design Audit - Stub Script
Referenced by: /mobile workflow

Checks mobile app projects for:
- Touch target sizes (44x44 minimum)
- Color contrast ratios
- Platform guidelines compliance
"""

import sys
import os
import json
from pathlib import Path


def audit_project(project_path: str) -> dict:
    """Audit mobile project for design issues."""
    results = {
        "project_path": project_path,
        "issues": [],
        "warnings": [],
        "passed": []
    }
    
    target = Path(project_path)
    
    # Detect platform
    platforms = []
    if (target / "android").exists() or (target / "app/build.gradle").exists():
        platforms.append("android")
    if (target / "ios").exists() or (target / "*.xcodeproj").exists():
        platforms.append("ios")
    if (target / "package.json").exists():
        pkg_path = target / "package.json"
        try:
            pkg = json.loads(pkg_path.read_text())
            deps = {**pkg.get("dependencies", {}), **pkg.get("devDependencies", {})}
            if "react-native" in deps:
                platforms.append("react-native")
            if "flutter" in str(pkg):
                platforms.append("flutter")
        except Exception:
            pass
    
    if (target / "pubspec.yaml").exists():
        platforms.append("flutter")
    
    if platforms:
        results["passed"].append(f"✅ Detected platforms: {', '.join(platforms)}")
    else:
        results["warnings"].append("⚠️ Could not detect mobile platform")
    
    # Check for accessibility files
    a11y_indicators = [
        "accessibility", "a11y", "contentDescription",
        "accessibilityLabel", "semanticLabel"
    ]
    
    source_files = list(target.rglob("*.tsx")) + list(target.rglob("*.jsx"))
    source_files += list(target.rglob("*.dart")) + list(target.rglob("*.kt"))
    source_files += list(target.rglob("*.swift"))
    
    has_a11y = False
    for sf in source_files[:20]:  # Sample first 20
        try:
            content = sf.read_text()
            if any(ind in content for ind in a11y_indicators):
                has_a11y = True
                break
        except Exception:
            pass
    
    if has_a11y:
        results["passed"].append("✅ Accessibility attributes detected")
    else:
        results["warnings"].append("⚠️ No accessibility attributes found in sampled files")
    
    # Check for touch target sizing
    touch_indicators = ["minHeight: 44", "minWidth: 44", "hitSlop", "touchableOpacity"]
    has_touch_targets = False
    for sf in source_files[:20]:
        try:
            content = sf.read_text()
            if any(ind in content for ind in touch_indicators):
                has_touch_targets = True
                break
        except Exception:
            pass
    
    if has_touch_targets:
        results["passed"].append("✅ Touch target sizing considerations found")
    else:
        results["warnings"].append("⚠️ Consider adding minimum touch target sizes (44x44)")
    
    if not results["issues"]:
        results["passed"].append("✅ Mobile design audit complete")
    
    return results


def print_results(results: dict):
    """Print audit results."""
    print("\n" + "=" * 50)
    print("📱 MOBILE DESIGN AUDIT")
    print("=" * 50)
    print(f"Project: {results['project_path']}\n")
    
    if results["issues"]:
        print("❌ ISSUES:")
        for issue in results["issues"]:
            print(f"  {issue}")
        print()
    
    if results["warnings"]:
        print("⚠️ WARNINGS:")
        for warning in results["warnings"]:
            print(f"  {warning}")
        print()
    
    if results["passed"]:
        print("✅ PASSED:")
        for passed in results["passed"]:
            print(f"  {passed}")
        print()
    
    if results["issues"]:
        print("❌ AUDIT FAILED")
        return 1
    else:
        print("✅ AUDIT PASSED")
        return 0


def main():
    if len(sys.argv) < 2:
        print("Usage: python mobile_audit.py <project_path>")
        sys.exit(1)
    
    project_path = sys.argv[1]
    if not os.path.exists(project_path):
        print(f"Error: Path '{project_path}' does not exist")
        sys.exit(1)
    
    results = audit_project(project_path)
    exit_code = print_results(results)
    sys.exit(exit_code)


if __name__ == "__main__":
    main()
